### PR TITLE
Update CHANGE_LOG for 0.56.0

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
@gmarkall any chance you could please provide `CHANGE_LOG` updates for 0.56 with regards to the CUDA target? Many thanks!

**Comment by stuartarchibald** [comment]
Thanks for 46b8cda @gmarkall

File: CHANGE_LOG [hunk 1]
```diff
@@ -104,7 +104,208 @@ General Enhancements:
...
+* `readthedocs-assistant <https://github.com/readthedocs-assistant>`_
+* `Rishi Kulkarni <https://github.com/rishi-kulkarni>`_
+* `Sterling Baird <https://github.com/sgbaird>`_
+* `Siu Kwan Lam <https://github.com/sklam>`_
+* `stuartarchibald <https://github.com/stuartarchibald>`_
+* `Stepan Rakitin <https://github.com/svrakitin>`_
+* `Michael Collison <https://github.com/testhound>`_
+* `Tingkai Liu <https://github.com/TK-21st>`_
+
+Total authors: 34
```
Comment (stuartarchibald) [inline CHANGE_LOG:308]:
```suggestion
```
This isn't usually included?

File: CHANGE_LOG [hunk 2]
```diff
@@ -104,7 +104,208 @@ General Enhancements:
...
+* PR `#8181 <https://github.com/numba/numba/pull/8181>`_: Fix various typos (`luzpaz <https://github.com/luzpaz>`_)
+* PR `#8187 <https://github.com/numba/numba/pull/8187>`_: Update CHANGE_LOG for 0.55.2 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_)
+* PR `#8189 <https://github.com/numba/numba/pull/8189>`_: updated version support information for 0.55.2/0.57 (`esc <https://github.com/esc>`_)
+* PR `#8191 <https://github.com/numba/numba/pull/8191>`_: CUDA: Update deprecation notes for 0.56. (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#8195 <https://github.com/numba/numba/pull/8195>`_: Make the workqueue threading backend once again fork safe. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8196 <https://github.com/numba/numba/pull/8196>`_: Fix numerical tolerance in parfors caching test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8197 <https://github.com/numba/numba/pull/8197>`_: Fix `isinstance` warning check test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+
+Total PRs: 159
```
Comment (stuartarchibald) [inline CHANGE_LOG:269]:
```suggestion
```
This isn't usually included?

**Comment by esc** [comment]
The docs won't build because readthedocs doesn't merge with `main` but instead uses the `head` of the current branch. That means changes from #8203 aren't being used for the doc build.

**Comment by esc** [comment]
> The docs won't build because readthedocs doesn't merge with `main` but instead uses the `head` of the current branch. That means changes from #8203 aren't being used for the doc build.

That is of course, somewhat unfortunate, since the docs build is what we are interested in...

**Comment by stuartarchibald** [comment]
> The docs won't build because readthedocs doesn't merge with `main` but instead uses the `head` of the current branch. That means changes from #8203 aren't being used for the doc build.

I merged in `main` to try and resolve this.

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,3 +1,309 @@
...
+  ``_dbg_optnone`` kwarg in the ``@jit`` decorator family).
+* A new environment variable is added ``NUMBA_EXTEND_VARIABLE_LIFETIMES``, which
+  if set will extend the lifetime of variables to the end of their basic block,
+  this to permit a debugging experience in GDB similar to that found in compiled
+  C/C++/Fortran code.
+
+NumPy features/enhancements:
+
+* Initial support for passing, using and returning ``numpy.random.Generator``
+  instances has been added, this includes some commonly used distributions.
```
Comment (esc) [inline CHANGE_LOG:45]: Since #8038 didn't make it, do we need to adapt this line?

Comment (stuartarchibald) [reply CHANGE_LOG]: Have changed it, f046462.

File: CHANGE_LOG [hunk 4]
```diff
@@ -1,3 +1,310 @@
+Version 0.56.0 RC1 (28 June, 2022)
+----------------------------------
+
+This release continues to add new features, bug fixes and stability improvements
+to Numba. Please note that this will be the last release that has support for
+Python 3.7 as the next release series (Numba 0.57) will support Python 3.11!
+
```
Comment (esc) [inline CHANGE_LOG:7]:
```suggestion
Also note that, this will be the last release to support linux-32 packages produced by the Numba team.

```

**Review by esc — DISMISSED** [review]
A few minor nitpicks, looks good otherwise.

File: CHANGE_LOG [hunk 5]
```diff
@@ -1,3 +1,309 @@
...
+  to use, single reference point, for looking up information about Numba
+  releases and their required/supported dependencies.
+
+General Enhancements:
+
+* Numba imports more quickly in environments with large numbers of packages as
+  it now uses ``importlib-metadata`` for querying other packages.
+* Emission of chrome tracing output is now supported for the internal
+  compilation event handling system.
+* This release is tested and known to work when using the Pyston python
```
[the remark this answers is not in the document]
Comment (esc) [reply CHANGE_LOG:105]: This needs a RTD render check once merged.

File: CHANGE_LOG [hunk 6]
```diff
@@ -1,3 +1,310 @@
...
+* PR `#8187 <https://github.com/numba/numba/pull/8187>`_: Update CHANGE_LOG for 0.55.2 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_)
+* PR `#8189 <https://github.com/numba/numba/pull/8189>`_: updated version support information for 0.55.2/0.57 (`esc <https://github.com/esc>`_)
+* PR `#8191 <https://github.com/numba/numba/pull/8191>`_: CUDA: Update deprecation notes for 0.56. (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#8195 <https://github.com/numba/numba/pull/8195>`_: Make the workqueue threading backend once again fork safe. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8196 <https://github.com/numba/numba/pull/8196>`_: Fix numerical tolerance in parfors caching test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8197 <https://github.com/numba/numba/pull/8197>`_: Fix `isinstance` warning check test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8203 <https://github.com/numba/numba/pull/8203>`_: pin llvmlite 0.39 for public CI builds (`esc <https://github.com/esc>`_)
+
```
Comment (esc) [inline CHANGE_LOG:270]:
```suggestion
PR `#8192 <https://github.com/numba/numba/pull/8192>`_:  Update CHANGE_LOG for 0.56.0 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`_)

```

**Comment by esc** [comment]
https://github.com/numba/numba/pull/8192/files#r908767941 <-- suggests to add this PR to itself (on the PR list).

**Comment by sklam** [comment]
I verified the list of PRs. Matches what i got

File: CHANGE_LOG [hunk 7]
```diff
@@ -1,3 +1,312 @@
...
+* PR `#8189 <https://github.com/numba/numba/pull/8189>`_: updated version support information for 0.55.2/0.57 (`esc <https://github.com/esc>`_)
+* PR `#8191 <https://github.com/numba/numba/pull/8191>`_: CUDA: Update deprecation notes for 0.56. (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#8195 <https://github.com/numba/numba/pull/8195>`_: Make the workqueue threading backend once again fork safe. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8196 <https://github.com/numba/numba/pull/8196>`_: Fix numerical tolerance in parfors caching test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8197 <https://github.com/numba/numba/pull/8197>`_: Fix `isinstance` warning check test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#8203 <https://github.com/numba/numba/pull/8203>`_: pin llvmlite 0.39 for public CI builds (`esc <https://github.com/esc>`_)
+PR `#8192 <https://github.com/numba/numba/pull/8192>`_:  Update CHANGE_LOG for 0.56.0 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (sklam) [inline CHANGE_LOG:271]:
```suggestion
* PR `#8192 <https://github.com/numba/numba/pull/8192>`_:  Update CHANGE_LOG for 0.56.0 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
```

File: CHANGE_LOG [hunk 8]
```diff
@@ -1,3 +1,312 @@
+Version 0.56.0 RC1 (28 June, 2022)
+----------------------------------
+
+This release continues to add new features, bug fixes and stability improvements
+to Numba. Please note that this will be the last release that has support for
+Python 3.7 as the next release series (Numba 0.57) will support Python 3.11!
+Also note that, this will be the last release to support linux-32 packages produced by the Numba team.
```
Comment (esc) [inline CHANGE_LOG]:
```suggestion
Also note that, this will be the last release to support linux-32 packages
produced by the Numba team.
```

**Review by esc — APPROVED** [review]

**Review by sklam — APPROVED** [review]
LGTM

**Pull request merged**